packages/convai-widget-core/src/widget/Transcript.tsx

```diff
@@ -0,0 +1,49 @@
+import { ReadonlySignal, Signal, useSignalEffect } from "@preact/signals";
+import { TranscriptEntry } from "../contexts/conversation";
+import { useEffect, useRef } from "preact/compat";
+import { TranscriptMessage } from "./TranscriptMessage";
+
+interface TranscriptProps {
+  scrollPinned: Signal<boolean>;
+  transcript: ReadonlySignal<TranscriptEntry[]>;
+}
+
+export function Transcript({ scrollPinned, transcript }: TranscriptProps) {
+  const scrollContainer = useRef<HTMLDivElement>(null);
+  const scrollToBottom = () => {
+    scrollContainer.current?.scrollTo(0, scrollContainer.current.scrollHeight);
+  };
```

Comment:
I haven't tried yet, so have no idea how it looks, but I feel like these scroll jumps/transitions influence a lot how the whole interaction feels. Would be great to polish the timings, especially if messages are sliding in or something.